packages/core/src/Room.ts

```diff
@@ -599,18 +614,23 @@ export abstract class Room<State extends object= any, Metadata= any> {
       }
     }
 
-    // emit 'join' to room handler
-    this._events.emit('join', client);
+    // emit 'join' to room handler (if not reconnecting)
+    if (!previousReconnectionToken) {
+      this._events.emit('join', client);
```

Comment:
here, if player has disconnected, the 'join' is being emitted after 'leave'

Reply:
since this is only for a new join, anyway, could put it right above where it checks for LEAVING

Reply:
You're right! Thanks for spotting this!